src/data/bucket/fill_extrusion_bucket.js

```diff
@@ -447,13 +447,24 @@ class FillExtrusionBucket implements Bucket {
 
                 let kPrev, kFirst;
 
+                let a, b, na, nb;
+                {
+                    const p0 = ring[0];
+                    const p1 = ring[1];
+                    a = p1.sub(p0);
```

Comment:
> I might be misreading the code here, but it seems that unlike the previous PR, this one is starting to go into the territory of diminishing returns. Essentially we're avoiding a single normal calculation per point (`p._perp()._unit()`), at the cost of quite a bit more code. And at this point it's not necessarily faster due to v8 optimizations — so unless profiling shows a clear improvement, I'd settle for clarity and simplicity in code instead.

I agree that the code has become bit harder to follow. I didn't do any profiling here but I think leaving `na` and `nb` while removing `a` and `b` should fix the issue you're referring to. I added these (`a` and `b`) only to avoid the `nEdge` calculations totally forgetting the fact that we're building new vectors which itself can be expensive and would counter the original motif. Let me know if this makes sense or you still think the whole change here is unnecessary perhaps I am missing something else. cc @mourner